pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -42,49 +45,68 @@ static __always_inline void get_tcp_segment_counts(struct sock* skp, __u32* pack
 
 SEC("kprobe/tcp_sendmsg")
 int kprobe__tcp_sendmsg(struct pt_regs* ctx) {
-    __u32 packets_in = 0;
-    __u32 packets_out = 0;
-    struct sock* skp = (struct sock*)PT_REGS_PARM1(ctx);
-    size_t size = (size_t)PT_REGS_PARM3(ctx);
     u64 pid_tgid = bpf_get_current_pid_tgid();
-    log_debug("kprobe/tcp_sendmsg: pid_tgid: %d, size: %d\n", pid_tgid, size);
-
-    conn_tuple_t t = {};
-    if (!read_conn_tuple(&t, skp, pid_tgid, CONN_TYPE_TCP)) {
-        return 0;
-    }
-
-    handle_tcp_stats(&t, skp);
-    get_tcp_segment_counts(skp, &packets_in, &packets_out);
-    return handle_message(&t, size, 0, CONN_DIRECTION_UNKNOWN, packets_out, packets_in, PACKET_COUNT_ABSOLUTE);
+    log_debug("kprobe/tcp_sendmsg: pid_tgid: %d\n", pid_tgid);
+    struct sock* parm1 = (struct sock*)PT_REGS_PARM1(ctx);
+    struct sock* skp = NULL;
+    bpf_probe_read(&skp, sizeof(skp), &parm1);
```

Comment:
I presume this is to satisfy the 4.4 verifier needing the value being on the stack? Did you try `struct sock *skp = parm1;`? That might be sufficient.

Reply:
Yes that worked.